test/jdk/java/net/httpclient/HttpResponseConnectionLabelTest.java

```diff
@@ -0,0 +1,444 @@
+
+        private ServerRequestPair {
+            try {
+                server.start();
+            } catch (Exception serverException) {
+                throw new RuntimeException("failed closing server", serverException);
+            }
+        }
+
+        private static ServerRequestPair of(Version version, boolean secure) {
+
+            // Create the server and the request URI
+            SSLContext sslContext = secure ? SSL_CONTEXT : null;
+            HttpTestServer server = createServer(version, sslContext);
+            String handlerPath = "/" + /* salting the path: */ HttpResponseConnectionLabelTest.class.getSimpleName();
+            String requestUriScheme = secure ? "https" : "http";
+            URI requestUri = URI.create(requestUriScheme + "://" + server.serverAuthority() + handlerPath);
+
+            // Register the request handler
+            String serverId = "" + SERVER_COUNTER.getAndIncrement();
+            CountDownLatch[] serverResponseLatchRef = new CountDownLatch[1];
+            server.addHandler(
+                    (exchange) -> {
+                        String responseBody = "" + SERVER_RESPONSE_COUNTER.getAndIncrement();
+                        String connectionKey = exchange.getConnectionKey();
+                        try (exchange) {
+
+                            // Participate in the latch count down
+                            if (serverResponseLatchRef[0] != null) {
+                                serverResponseLatchRef[0].countDown();
+                                LOGGER.log(
+                                        "Server[%s] is waiting for the latch... (connectionKey=%s, responseBody=%s)",
+                                        serverId, connectionKey, responseBody);
+                                serverResponseLatchRef[0].await();
+                            }
+
+                            // Write the response
+                            LOGGER.log(
+                                    "Server[%s] is responding... (connectionKey=%s, responseBody=%s)",
+                                    serverId, connectionKey, responseBody);
+                            exchange.getResponseHeaders().addHeader(CONNECTION_KEY_HEADER_NAME, connectionKey);
+                            exchange.getResponseHeaders().addHeader(SERVER_ID_HEADER_NAME, serverId);
+                            byte[] responseBodyBytes = responseBody.getBytes(CHARSET);
+                            exchange.sendResponseHeaders(200, responseBodyBytes.length);
+                            exchange.getResponseBody().write(responseBodyBytes);
+
+                        } catch (IOException ioe) {
+                            String message = "Server[%s] has failed! (connectionKey=%s, responseBody=%s)"
+                                    .formatted(serverId, connectionKey, responseBody);
+                            throw new RuntimeException(message, ioe);
```

Comment:
I would recommend logging something here before throwing the exception. For instance calling `ioe.printStackTrace()`. There's not supposed to be one, and if there  is one it's not guaranteed that throwing it will ensure that it gets logged. The client might fail with some other error, or the test may timeout, and we will never know what happened.

Reply:
Implemented to both log and throw in a0859c7c68614e8c3f531b1d4d9357f3f48cf5ab.